Debianpackage

carpoon opened the pull request:
Changes mostly related to debian packaging and some recent updates.

carpoon committed:
prepare for debian package

* Add make file
* Add debian folder
* Patch for the changed funcions lib location
* Fix some references from /etc/firewall to /etc/iptablez

carpoon committed:
Add local changes

Add local changes
Fix symlink typo
Bump package version

akorn commented on xt_geoip/update.zsh:
Is `xtables-addons-dkms` really needed by the geoip update script? I don't think so.

carpoon replied:
No, the `xtables-addons-common` package should be enough.

akorn commented on debian/control:
Please make the `Maintainer` something like `Maintainer: Unofficial Build <please-set-a-maintainer@example.com>` -- people can fork this repository and create their own packages, from potentially modified sources. I don't want to be listed as the maintainer of those.

When we build the package, we should set ourselves to be the maintainer of the respective binary package we create.

akorn commented on debian/copyright:
Since you're submitting the result of a nontrivial amount of work, please add yourself as a copyright holder.

akorn reviewed (changes requested):
I don't like that you're just replacing one hardcoded path (`/etc/firewall`) with another (`/etc/iptablez`), as this is an incompatible change.

Even though I agree with the need to move from the generic `firewall` to the more specific `iptablez`, I don't think this is a good way of handling that.

Either these paths should be configurable somehow (although that's unlikely to be useful in the long term, because it's sensible to use `/etc/iptablez` if that's what the project is called); or -- perhaps better -- let the code fall back to `/{etc,var/lib}/firewall` if the corresponding `iptablez` directory doesn't exist yet, and print an annoying warning.

carpoon committed:
Add myself as contributor

carpoon committed:
Fix dependency

Fix dependency as only the xtables-addons-common package is needed to
download and process the geoip data.
Only use an example for maintner for the debian package.

akorn commented:
Thanks; I merged the PR. I'll add the compatibility/nagging code for the directory change later (or, of course, feel free to submit another PR that adds it).
